extension/src/popup/ducks/transactionSubmission.ts

```diff
@@ -260,6 +276,27 @@ const transactionSubmissionSlice = createSlice({
         assetIcons,
       };
     });
+    builder.addCase(getBestPath.fulfilled, (state, action) => {
+      if (!action.payload) {
+        state.transactionData.path = [];
+        state.transactionData.destinationAmount = "";
+        return;
+      }
+
+      // store in canonical form for easier use
+      const path = [];
+      for (let i = 0; i < action.payload.path.length; i += 1) {
```

Comment:
there's nothing really wrong with this, but you could use `.forEach` here to simplify this code a little

Reply:
much cleaner nice https://github.com/stellar/freighter/pull/413/commits/4ae8aab7fc729ce7c62f9cef7848ba7abec19e48